openapi-validation-core/src/main/java/com/getyourguide/openapi/validation/core/OpenApiRequestValidator.java

```diff
@@ -79,14 +81,22 @@ public ValidationResult validateRequestObject(final RequestMetaData request, Str
     private static SimpleRequest buildSimpleRequest(RequestMetaData request, String requestBody) {
         var requestBuilder = new SimpleRequest.Builder(request.getMethod(), request.getUri().getPath());
         URLEncodedUtils.parse(request.getUri(), StandardCharsets.UTF_8)
-            .forEach(p -> requestBuilder.withQueryParam(p.getName(), p.getValue()));
+            .forEach(p -> requestBuilder.withQueryParam(p.getName(), getQueryParameterValueWithOptionalDecode(p)));
         if (requestBody != null) {
             requestBuilder.withBody(requestBody);
         }
         request.getHeaders().forEach(requestBuilder::withHeader);
         return requestBuilder.build();
     }
 
+    private static String getQueryParameterValueWithOptionalDecode(NameValuePair p) {
+        if (p.getValue() == null) {
+            return null;
+        }
+
+        return URLDecoder.decode(p.getValue(), StandardCharsets.UTF_8);
```

Comment:
Is UTF-8 known to be the correct charset? Should this possibly be funneled through from the request headers?

Reply:
Yes, UTF-8 is correct. There is no header for encoding of URI as that is defined per the standard. Only encoding defined in headers is the content (body).

From https://www.rfc-editor.org/rfc/rfc3986
> When a new URI scheme defines a component that represents textual data consisting of characters from the Universal Character Set [[UCS](https://www.rfc-editor.org/rfc/rfc3986#ref-UCS)], the data should first be encoded as octets according to the UTF-8 character encoding [[STD63](https://www.rfc-editor.org/rfc/rfc3986#ref-STD63)]; then only those octets that do not correspond to characters in the unreserved set should be percent-encoded.  For example, the character A would be represented as "A", the character LATIN CAPITAL LETTER A WITH GRAVE would be represented as "%C3%80", and the character KATAKANA LETTER A would be represented as "%E3%82%A2".

Reply:
Thanks for the clarification!